File: 1337-the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        set<pair<int, int>> se1;
        for(int i=0; i<mat.size(); i++){
            int tmp = accumulate(mat[i].begin(), mat[i].end(), 0);
            se1.insert({tmp, i});
        }
        vector<int> ans;
        for(auto it = se1.begin(); k>0; it++, k--){
            ans.push_back(it->second);
        }
        return ans;
    }
};

File: 1337-the-k-weakest-rows-in-a-matrix.py
class Solution(object):
    def kWeakestRows(self, mat, k):

        # tmp là mảng lưu lại số soldiers của mỗi hàng
        # stores numbers of soldiers of each row
        tmp = []

        for x in mat:
            tmp.append(sum(x))
        if len(tmp) == 0: 
            return [0]
        
        ans = []
        # cho mỗi giá tri j trong sorted(tmp) thì append index của j trong tmp vào ans
        for j in sorted(tmp):
            a = tmp.index(j) # returns the position at the first occurrence of the specified value.
            ans.append(a)
            
            # cho tmp[a] = -1 để hàm index() không lấy lại vị trí này nữa, 
            # có thể cho bằng số bất kì
            tmp[a] = -1
        
        # ans[:k] trả về chuỗi con có k phần tử đầu của ans
        return ans[:k]
